### PR TITLE
Use `InferType` for type inference and drop usage of deprecated functions

### PR DESCRIPTION
**Pull request by blitz-1306**
## Preface

This PR acts as an End-to-End test for a **solc-typed-ast** package with changes from https://github.com/ConsenSys/solc-typed-ast/pull/159. Changes in **solc-typed-ast** would cause backward compatibility break, so we want to ensure that consumers are able to move logic on top of new implementation.

## Significant changes
- [x] Drop parsing types in macro variables. They are not used anyway, Scribble needs just names. This seem to be only place where Scribble relies on typeString parser (that is considered as **deprecated** in **solc-typed-ast** for now at least).
- [x] Inject `InferType` everywhere, where necessary.
- [x] Update dependencies (currently use local **solc-typed-ast** build).

## Concerns
- Mapping interposition is currently affected due to it seems to also rely on typeString parser instead of AST. The AST is destructively modified by `replaceNode()` so AST-based type inference can not function properly. I guess another approach is needed here.
- There are cleanup **TODO**s, what worth addressing. Most of them are related to reworking functionality, that is now provided by **solc-typed-ast**.

Regards.

### CODE REVIEW

**Review by cd1m0 — CHANGES_REQUESTED** [review]
Good work! All the @todos are on point. Taking lock on PR to work on these

File: src/instrumenter/instrument.ts [hunk 1]
```diff
@@ -103,51 +104,20 @@ export function changesMutability(ctx: InstrumentationContext): boolean {
...
-        }
-
-        res.push(call);
-    }
-
-    return res;
+export function findExternalCalls(node: ContractDefinition | FunctionDefinition): FunctionCall[] {
+    const interestingExternalCallBuiltins = ["call", "delegatecall", "staticcall"];
+
+    return node.getChildrenBySelector(
```
[the remark this answers is not in the document]
Comment (blitz-1306) [reply src/instrumenter/instrument.ts:113]: Well, it is actually the same as it was, with few twists... Literally, get childred, where:
- child is a `FunctionCall`
- child is not an instrumentation node (`typeString` is `<missing>`)
- child is an [external call by **solc-typed-ast**'s opinion](https://github.com/ConsenSys/solc-typed-ast/blob/ece08a453a562662b82431092695446ffec6b192/src/types/utils.ts#L351-L368)
- child function call refers to a user-defined function
  - or (if it is a builtin) is only refers to `call`, `delegatecall` or `staticcall` (due to **solc-typed-ast** considers [a borader list of builtins](https://github.com/ConsenSys/solc-typed-ast/blob/ece08a453a562662b82431092695446ffec6b192/src/types/utils.ts#L66) as external call candidates)

So, basically it is not that different than what we had before, except that is more compact.

[the remark this answers is not in the document]
Comment (cd1m0) [reply src/instrumenter/instrument.ts:113]: Ok. Not strongly attached to changing it.

**Review by blitz-1306 — COMMENTED** [review]
Looks good. I would rebase other branch to see if we can bump coverage.

**Comment by blitz-1306** [comment]
I rebased #203. Test coverage there is passong required thresholds. We can ether merge it, or I can apply limited set of changes to this branch manually. WDYT?

**Review by cd1m0 — APPROVED** [review]
